jdbc/src/java/org/apache/hive/jdbc/HiveStatement.java

```diff
@@ -587,6 +587,26 @@ public int getUpdateCount() throws SQLException {
     return (int) numModifiedRows;
   }
 
+  @Override
+  public long getLargeUpdateCount() throws SQLException {
+    checkConnection("getLargeUpdateCount");
+    /**
+     * Poll on the operation status, till the operation is complete. We want to ensure that since a
+     * client might end up using executeAsync and then call this to check if the query run is
+     * finished.
+     */
+    long numModifiedRows = -1L;
+    TGetOperationStatusResp resp = waitForOperationToComplete();
+    if (resp != null) {
+      numModifiedRows = resp.getNumModifiedRows();
+    }
+    if (numModifiedRows == -1L || numModifiedRows > Long.MAX_VALUE) {
```

Comment:
is `-2` valid?
we could reuse the newly implemented method in the old `getUpdateCount` to reduce code duplication